### DISCUSSION TITLE
Spacing between scalar bar title and ticks

### DISCUSSION THREAD

**Discussion by danielfails** [pyvista/pyvista#4668] [Q&A] [answered]
Hello everyone,

I would like to adjust the spacing between the title of the scalar bar and its ticks (red line in pic). Unfortunately, I haven't found a way to do this yet. I've already tried adjusting the height, font size, and such, but none of it worked. `plotter.scalar_bar.SetVerticalTitleSeparation(10)` also had no effect.

Can someone help me? :)

```
import pyvista as pv
sphere = pv.Sphere()
sphere['Data'] = sphere.points[:, 2]
plotter = pv.Plotter()
_ = plotter.add_mesh(sphere, show_scalar_bar=False)
_ = plotter.add_scalar_bar(
    'good looking title',
    interactive=True,
    vertical=False,
    title_font_size=35,
    label_font_size=30,
    fmt='%10.5f'
)

plotter.show()
```
![title](https://github.com/pyvista/pyvista/assets/132590594/1bb72e45-b4db-42be-b9d6-11bf816aa887)

**Comment by darikg** [comment, marked as answer]
You can get the `vtkScalarBarActor` with `bar = plotter.add_scalar_bar(...)` (edit: or use `plotter.scalar_bar` like you're already doing.)

The [`SetVerticalTitleSeparation`](https://vtk.org/doc/nightly/html/classvtkScalarBarActor.html#a55a8e055bf26a71ca02ad94507b067ad) method apparently only does anything when the bar is oriented vertically.

One kludge I found instead for the horizontal bar is to add a newline character to the title text, and then set the line spacing to refine the distance:
```python
bar = plotter.add_scalar_bar('good looking title\n', vertical=False, ...)
bar.GetTitleTextProperty().SetLineSpacing(1.5)
```

**Reply by danielfails** [reply]
Great answer, thank you! :)
